golangci-lint fails due to formatting issues in connectors code

Result of `make verify` includes:

```
connectors/opa/main.go:22:29: commentFormatting: put a space between `//` and comment text (gocritic)
const defaultPort = "50082" //synched with opa_connector.yaml
                            ^
connectors/opa/main.go:41:40: `defualt` is a misspelling of `default` (misspell)
                log.Printf("Env. variable not found, defualt value used: %s - %s\n", key, defaultValue)
                                                     ^
connectors/opa/main.go:49: unnecessary leading newline (whitespace)
func (s *server) GetPoliciesDecisions(ctx context.Context, in *pb.ApplicationContext) (*pb.PoliciesDecisions, error) {

connectors/opa/lib/opa_reader.go:41:2: commentFormatting: put a space between `//` and comment text (gocritic)
        //to store the list of DatasetDecision
        ^
connectors/opa/lib/opa_reader.go:54:3: commentFormatting: put a space between `//` and comment text (gocritic)
                //Encode operation in a map[string]interface
                ^
connectors/opa/lib/catalog_reader.go:60:1: receiver name c should be consistent with previous receiver name r for CatalogReader (golint)
func (c *CatalogReader) GetDatasetMetadata(ctx *context.Context, client pb.DataCatalogServiceClient, datasetID string, appID string) (map[string]interface{}, error) {
^
connectors/opa/lib/opa_reader_test.go:24:36: var `opaServerUrl` should be `opaServerURL` (golint)
        timeOutSecs, catalogConnectorURL, opaServerUrl := tu.GetEnvironment()
                                          ^
connectors/opa/lib/opa_reader_test.go:22: unnecessary leading newline (whitespace)
func TestMainOpaConnector(t *testing.T) {

connectors/vault/vaultconn-bl/vault_reader_test.go:52:2: S1021: should merge variable declaration with assignment on next line (gosimple)
        var vault vltutils.VaultConnection
        ^
connectors/opa/testutil/helper.go:34:2: var `opaServerUrl` should be `opaServerURL` (golint)
        opaServerUrl := EnvValues["OPA_SERVER_URL"]
        ^
connectors/opa/testutil/helper.go:139:21: func parameter `appId` should be `appID` (golint)
func GetCatalogInfo(appId string, datasetID string) *pb.CatalogDatasetInfo {
                    ^
connectors/opa/testutil/helper.go:85:3: ineffectual assignment to `operation` (ineffassign)
                operation := datasetContext.GetOperation()
                ^
connectors/opa/testutil/helper.go:141:2: S1021: should merge variable declaration with assignment on next line (gosimple)
        var componentsMetadata map[string]*pb.DataComponentMetadata
        ^
connectors/opa/testutil/helper.go:145:2: S1021: should merge variable declaration with assignment on next line (gosimple)
        var datasetNamedMetadata map[string]string
        ^
connectors/vault/vault_connector.go:46:15: ineffectual assignment to `err` (ineffassign)
        timeOutSecs, err := strconv.Atoi(timeOutInSecs)
```

--- a/connectors/opa/lib/catalog_reader.go
+++ b/connectors/opa/lib/catalog_reader.go
@@ -57,17 +57,17 @@ func (r *CatalogReader) GetDatasetsMetadataFromCatalog(in *pb.ApplicationContext
 	return datasetsMetadata, nil
 }
 
-func (c *CatalogReader) GetDatasetMetadata(ctx *context.Context, client pb.DataCatalogServiceClient, datasetID string, appID string) (map[string]interface{}, error) {
+func (r *CatalogReader) GetDatasetMetadata(ctx *context.Context, client pb.DataCatalogServiceClient, datasetID string, appID string) (map[string]interface{}, error) {
 	objToSend := &pb.CatalogDatasetRequest{AppId: appID, DatasetId: datasetID}
 	log.Printf("Sending request to External Catalog Connector: datasetID = %s", datasetID)
-	r, err := client.GetDatasetInfo(*ctx, objToSend)
+	info, err := client.GetDatasetInfo(*ctx, objToSend)
 	if err != nil {
 		return nil, fmt.Errorf("error sending data to External Catalog Connector (datasetID = %s): %v", datasetID, err)
 	}
 
 	log.Printf("Received Response from External Catalog Connector for  dataSetID: %s\n", datasetID)
 	log.Printf("Response received from External Catalog Connector is given below:")
-	responseBytes, errJSON := json.MarshalIndent(r, "", "\t")
+	responseBytes, errJSON := json.MarshalIndent(info, "", "\t")
 	if errJSON != nil {
 		return nil, fmt.Errorf("error Marshalling External Catalog Connector Response: %v", errJSON)
 	}

--- a/connectors/opa/lib/opa_reader.go
+++ b/connectors/opa/lib/opa_reader.go
@@ -38,7 +38,7 @@ func (r *OpaReader) GetOPADecisions(in *pb.ApplicationContext, catalogReader *Ca
 		return nil, fmt.Errorf("error in unmarshalling appInfoBytes: %v", err)
 	}
 
-	//to store the list of DatasetDecision
+	// to store the list of DatasetDecision
 	var datasetDecisionList []*pb.DatasetDecision
 	for i, datasetContext := range in.GetDatasets() {
 		dataset := datasetContext.GetDataset()
@@ -51,7 +51,7 @@ func (r *OpaReader) GetOPADecisions(in *pb.ApplicationContext, catalogReader *Ca
 		}
 
 		operation := datasetContext.GetOperation()
-		//Encode operation in a map[string]interface
+		// Encode operation in a map[string]interface
 		operationBytes, err := json.MarshalIndent(operation, "", "\t")
 		log.Println("Operation Bytes: " + string(operationBytes))
 		if err != nil {
@@ -69,11 +69,11 @@ func (r *OpaReader) GetOPADecisions(in *pb.ApplicationContext, catalogReader *Ca
 				inputMap[k] = v
 			}
 		}
-		//Combine with appInfoMap
+		// Combine with appInfoMap
 		for k, v := range appInfoMap {
 			inputMap[k] = v
 		}
-		//Printing the combined map
+		// Printing the combined map
 		toPrintBytes, _ := json.MarshalIndent(inputMap, "", "\t")
 		log.Println("********sending this to OPA : *******")
 		log.Println(string(toPrintBytes))
@@ -87,10 +87,10 @@ func (r *OpaReader) GetOPADecisions(in *pb.ApplicationContext, catalogReader *Ca
 		if err != nil {
 			return nil, fmt.Errorf("error in GetOPAOperationDecision (i = %d): %v", i, err)
 		}
-		//add to a list
+		// Add to a list
 		var opaOperationDecisionList []*pb.OperationDecision
 		opaOperationDecisionList = append(opaOperationDecisionList, opaOperationDecision)
-		//Create a new *DatasetDecision
+		// Create a new *DatasetDecision
 		datasetDecison := &pb.DatasetDecision{Dataset: dataset, Decisions: opaOperationDecisionList}
 		datasetDecisionList = append(datasetDecisionList, datasetDecison)
 	}
@@ -154,7 +154,7 @@ func GetOPAOperationDecision(opaEval string, operation *pb.AccessOperation) (*pb
 		}
 	}
 
-	if len(enforcementActions) == 0 { //allow action
+	if len(enforcementActions) == 0 { // allow action
 		newEnforcementAction := &pb.EnforcementAction{Name: "Allow", Id: "Allow-ID", Level: pb.EnforcementAction_DATASET, Args: map[string]string{}}
 		enforcementActions = append(enforcementActions, newEnforcementAction)
 	}

--- a/connectors/opa/lib/opa_reader_test.go
+++ b/connectors/opa/lib/opa_reader_test.go
@@ -20,11 +20,10 @@ import (
 // format into enforcement decisions format
 
 func TestMainOpaConnector(t *testing.T) {
-
-	timeOutSecs, catalogConnectorURL, opaServerUrl := tu.GetEnvironment()
+	timeOutSecs, catalogConnectorURL, opaServerURL := tu.GetEnvironment()
 	applicationContext := tu.GetApplicationContext("marketing")
 
-	srv := NewOpaReader(opaServerUrl)
+	srv := NewOpaReader(opaServerURL)
 	catalogReader := NewCatalogReader(catalogConnectorURL, timeOutSecs)
 	policiesDecisions, err := srv.GetOPADecisions(applicationContext, catalogReader)
 	assert.NilError(t, err)

--- a/connectors/opa/main.go
+++ b/connectors/opa/main.go
@@ -19,7 +19,7 @@ import (
 
 var opaServerURL = ""
 
-const defaultPort = "50082" //synched with opa_connector.yaml
+const defaultPort = "50082" // synced with opa_connector.yaml
 
 type server struct {
 	pb.UnimplementedPolicyManagerServiceServer
@@ -38,7 +38,7 @@ func getEnv(key string) string {
 func getEnvWithDefault(key string, defaultValue string) string {
 	value, exists := os.LookupEnv(key)
 	if !exists {
-		log.Printf("Env. variable not found, defualt value used: %s - %s\n", key, defaultValue)
+		log.Printf("Env. variable not found, default value used: %s - %s\n", key, defaultValue)
 		return defaultValue
 	}
 
@@ -47,7 +47,6 @@ func getEnvWithDefault(key string, defaultValue string) string {
 }
 
 func (s *server) GetPoliciesDecisions(ctx context.Context, in *pb.ApplicationContext) (*pb.PoliciesDecisions, error) {
-
 	log.Println("Received ApplicationContext")
 	log.Println(in)
 	catalogConnectorAddress := getEnv("CATALOG_CONNECTOR_URL")

--- a/connectors/opa/testutil/helper.go
+++ b/connectors/opa/testutil/helper.go
@@ -6,12 +6,13 @@ package testutil
 import (
 	"context"
 	"fmt"
-	"github.com/ibm/the-mesh-for-data/manager/controllers/utils"
 	"log"
 	"net"
 	"net/http"
 	"strconv"
 	"testing"
+
+	"github.com/ibm/the-mesh-for-data/manager/controllers/utils"
 
 	"github.com/golang/protobuf/proto"
 	"github.com/stretchr/testify/assert"
@@ -31,10 +32,10 @@ func GetEnvironment() (int, string, string) {
 	timeOutSecs, _ := strconv.Atoi(timeOutInSecs)
 
 	catalogConnectorURL := EnvValues["CATALOG_CONNECTOR_URL"]
-	opaServerUrl := EnvValues["OPA_SERVER_URL"]
+	opaServerURL := EnvValues["OPA_SERVER_URL"]
 	log.Printf("EnvVariables = %v\n", EnvValues)
 
-	return timeOutSecs, catalogConnectorURL, opaServerUrl
+	return timeOutSecs, catalogConnectorURL, opaServerURL
 }
 
 func GetApplicationContext(purpose string) *pb.ApplicationContext {
@@ -69,11 +70,6 @@ func ConstructRedactColumn(colName string) *pb.EnforcementAction {
 		Level: pb.EnforcementAction_COLUMN, Args: map[string]string{"column_name": colName}}
 }
 
-/*func ConstructAllow() *pb.EnforcementAction {
-	return &pb.EnforcementAction{Name: "Allow", Id: "Allow-ID",
-		Level: pb.EnforcementAction_DATASET, Args: map[string]string{}}
-}*/
-
 func EnsureDeepEqualDecisions(t *testing.T, testedDecisions *pb.PoliciesDecisions, expectedDecisions *pb.PoliciesDecisions) {
 	assert.True(t, proto.Equal(testedDecisions, expectedDecisions), "Decisions we got from policyManager are not as expected. Expected: %v, Received: %v", expectedDecisions, testedDecisions)
 }
@@ -85,25 +81,19 @@ func GetExpectedOpaDecisions(purpose string, in *pb.ApplicationContext) *pb.Poli
 		operation := datasetContext.GetOperation()
 		enforcementActions := make([]*pb.EnforcementAction, 0)
 		usedPolicies := make([]*pb.Policy, 0)
-		operation = datasetContext.GetOperation()
 		fmt.Println("operation")
 		fmt.Println(operation)
 		var newUsedPolicy *pb.Policy
 		if purpose == "marketing" {
-			//newEnforcementAction := ConstructRedactColumn("nameDest")
 			newEnforcementAction := ConstructEncryptColumn("nameDest")
 			enforcementActions = append(enforcementActions, newEnforcementAction)
-			//newUsedPolicy = &pb.Policy{Description: "redact columns with name nameOrig and nameDest in datasets which have been tagged with Finance"}
 			newUsedPolicy = &pb.Policy{Description: "test for transactions dataset that encrypts some columns by name"}
 			usedPolicies = append(usedPolicies, newUsedPolicy)
 
-			//newEnforcementAction = ConstructRedactColumn("nameOrig")
 			newEnforcementAction = ConstructEncryptColumn("nameOrig")
 			enforcementActions = append(enforcementActions, newEnforcementAction)
-			//newUsedPolicy = &pb.Policy{Description: "redact columns with name nameOrig and nameDest in datasets which have been tagged with Finance"}
 			newUsedPolicy = &pb.Policy{Description: "test for transactions dataset that encrypts some columns by name"}
 			usedPolicies = append(usedPolicies, newUsedPolicy)
-			// newUsedPolicy = &pb.Policy{Description: "reduct columns with name nameOrig and nameDest  in datasets with Finance"}
 		} else {
 			newEnforcementAction := ConstructRemoveColumn("nameDest")
 			enforcementActions = append(enforcementActions, newEnforcementAction)
@@ -114,9 +104,7 @@ func GetExpectedOpaDecisions(purpose string, in *pb.ApplicationContext) *pb.Poli
 			enforcementActions = append(enforcementActions, newEnforcementAction)
 			newUsedPolicy = &pb.Policy{Description: "remove columns with name nameOrig and nameDest in datasets which have been tagged with Finance"}
 			usedPolicies = append(usedPolicies, newUsedPolicy)
-			// newUsedPolicy = &pb.Policy{Description: "remove columns with name nameOrig and nameDest  in datasets with Finance"}
 		}
-		//usedPolicies = append(usedPolicies, newUsedPolicy)
 		operationDecision := &pb.OperationDecision{Operation: operation,
 			EnforcementActions: enforcementActions,
 			UsedPolicies:       usedPolicies}
@@ -125,8 +113,7 @@ func GetExpectedOpaDecisions(purpose string, in *pb.ApplicationContext) *pb.Poli
 	return &pb.PoliciesDecisions{DatasetDecisions: []*pb.DatasetDecision{datasetDecison}}
 }
 
-/****************************/
-//Mocks for catalog-connector and OPA-connector
+// Mocks for catalog-connector and OPA-connector
 
 type connectorMockCatalog struct {
 	pb.UnimplementedDataCatalogServiceServer
@@ -136,14 +123,12 @@ func (s *connectorMockCatalog) GetDatasetInfo(ctx context.Context, req *pb.Catal
 	return GetCatalogInfo(req.GetAppId(), req.GetDatasetId()), nil
 }
 
-func GetCatalogInfo(appId string, datasetID string) *pb.CatalogDatasetInfo {
+func GetCatalogInfo(appID string, datasetID string) *pb.CatalogDatasetInfo {
 	var datasetInfo *pb.CatalogDatasetInfo
-	var componentsMetadata map[string]*pb.DataComponentMetadata
-	componentsMetadata = make(map[string]*pb.DataComponentMetadata)
+	componentsMetadata := make(map[string]*pb.DataComponentMetadata)
 	componentMetaData1 := &pb.DataComponentMetadata{}
 	componentsMetadata["first"] = componentMetaData1
-	var datasetNamedMetadata map[string]string
-	datasetNamedMetadata = make(map[string]string)
+	datasetNamedMetadata := make(map[string]string)
 
 	var datasetDetails *pb.DatasetDetails
 	datasetTags := []string{"Tag1"}
@@ -184,8 +169,6 @@ func MockCatalogConnector(port int) {
 
 func customOpaResponse(w http.ResponseWriter, r *http.Request) {
 	fmt.Println("Endpoint Hit: customOpaResponse")
-
-	//customeResponse := "{\"result\":{\"allow\":true,\"allowed_access_types\":[\"READ\",\"COPY\",\"WRITE\"],\"allowed_copy_destinations\":[\"NorthAmerica\",\"US\"],\"allowed_purposes\":[\"analysis\",\"fraud-detection\"],\"allowed_roles\":[\"DataScientist\",\"Security\"],\"deny\":[],\"transform\":[{\"arguments\":{\"column name\":\"nameDest\"},\"action_name\":\"redact column\",\"used_policy\":{\"description\":\"redact columns with name nameOrig and nameDest in datasets which have been tagged with Finance\"}},{\"arguments\":{\"column name\":\"nameOrig\"},\"action_name\":\"redact column\",\"used_policy\":{\"description\":\"redact columns with name nameOrig and nameDest in datasets which have been tagged with Finance\"}}]}}"
 
 	customeResponse := "{\"result\":{\"deny\":[],\"transform\":[{\"action_name\":\"encrypt column\",\"arguments\":{\"column_name\":\"nameDest\"},\"description\":\"Single column is encrypted with its own key\",\"used_policy\":{\"description\":\"test for transactions dataset that encrypts some columns by name\"}},{\"action_name\":\"encrypt column\",\"arguments\":{\"column_name\":\"nameOrig\"},\"description\":\"Single column is encrypted with its own key\",\"used_policy\":{\"description\":\"test for transactions dataset that encrypts some columns by name\"}}]}}"
 

--- a/connectors/vault/vault_connector.go
+++ b/connectors/vault/vault_connector.go
@@ -17,7 +17,7 @@ import (
 )
 
 const defaultTimeout = "180"
-const defaultPort = "50083" //synched with vault_connector.yaml
+const defaultPort = "50083" // synced with vault_connector.yaml
 
 type server struct {
 	pb.UnimplementedDataCredentialServiceServer
@@ -44,6 +44,9 @@ func main() {
 	vaultAddress := vaultutils.GetEnv(vaultutils.VaultAddressKey)
 	timeOutInSecs := vaultutils.GetEnvWithDefault(vaultutils.VaultTimeoutKey, defaultTimeout)
 	timeOutSecs, err := strconv.Atoi(timeOutInSecs)
+	if err != nil {
+		log.Fatalf("Environment variable %s is not an integer: %v", vaultutils.VaultTimeoutKey, err)
+	}
 	port := vaultutils.GetEnvWithDefault(vaultutils.VaultConnectorPortKey, defaultPort)
 
 	log.Printf("Vault address env variable in %s: %s\n", vaultutils.VaultAddressKey, vaultAddress)

--- a/connectors/vault/vault_utils/vault_utils.go
+++ b/connectors/vault/vault_utils/vault_utils.go
@@ -22,7 +22,7 @@ const (
 	VaultPathKey          string = "USER_VAULT_PATH"
 	VaultConnectorPortKey string = "PORT_VAULT_CONNECTOR"
 	DefaultTimeout        string = "180"
-	DefaultPort           string = "50083" //synched with vault_connector.yaml
+	DefaultPort           string = "50083" // synced with vault_connector.yaml
 )
 
 func GetEnv(key string) string {

--- a/connectors/vault/vaultconn-bl/vault_reader_test.go
+++ b/connectors/vault/vaultconn-bl/vault_reader_test.go
@@ -23,7 +23,6 @@ func TestVaultConnectorNormalRun(t *testing.T) {
 
 	srv := &Server{}
 	appID := "mock-appID"
-	//datasetID := "{\"ServerName\":\"cocoMDS3\",\"AssetGuid\":\"f6d9bf8c-dd37-4747-bca9-8ca9d1a5bb8f\"}"
 	datasetID := "mock-datasetID"
 	objToSendForCredential := &pb.DatasetCredentialsRequest{AppId: appID, DatasetId: datasetID}
 
@@ -49,11 +48,9 @@ func TestVaultConnectorNormalRun(t *testing.T) {
 
 	connection.Client = client
 	userVaultPath := tu.GetEnvironment()
-	var vault vltutils.VaultConnection
-	vault = connection
 
 	fmt.Println("userVaultPath in vault_connector_test: ", userVaultPath)
-	credentialsInfo, _ := srv.GetCredentialsInfo(objToSendForCredential, vault, userVaultPath)
+	credentialsInfo, _ := srv.GetCredentialsInfo(objToSendForCredential, connection, userVaultPath)
 	expectedCredentials := tu.GetExpectedVaultCredentials(objToSendForCredential)
 	tu.EnsureDeepEqualCredentials(t, credentialsInfo, expectedCredentials)
 }

--- a/hack/make-rules/verify.mk
+++ b/hack/make-rules/verify.mk
@@ -74,8 +74,7 @@ GOLINT_LINTERS ?= \
 CODE_MAINT += lint
 .PHONY: lint
 lint: $(TOOLBIN)/golangci-lint
-	# FIXME: currently passing the test even so there are linting issues
-	$(TOOLBIN)/golangci-lint run ${GOLINT_LINTERS} ./... ||  true
+	$(TOOLBIN)/golangci-lint run ${GOLINT_LINTERS} ./...
 
 .PHONY: lint-fix
 lint-fix: $(TOOLBIN)/golangci-lint

--- a/hack/make-rules/verify.mk
+++ b/hack/make-rules/verify.mk
@@ -74,7 +74,7 @@ GOLINT_LINTERS ?= \
 CODE_MAINT += lint
 .PHONY: lint
 lint: $(TOOLBIN)/golangci-lint
-	$(TOOLBIN)/golangci-lint run ${GOLINT_LINTERS} ./...
+	$(TOOLBIN)/golangci-lint run ${GOLINT_LINTERS} --timeout=5m ./...
 
 .PHONY: lint-fix
 lint-fix: $(TOOLBIN)/golangci-lint

